python/ray/train/jax/jax_trainer.py

```diff
@@ -0,0 +1,163 @@
+    configured for distributed jax training.
+
+    The ``train_loop_per_worker`` function is expected to take in either 0 or 1
+    arguments:
+
+    .. code-block:: python
+
+        def train_loop_per_worker():
+            ...
+
+    .. code-block:: python
+
+        def train_loop_per_worker(config: Dict):
+            ...
+
+    If ``train_loop_per_worker`` accepts an argument, then
+    ``train_loop_config`` will be passed in as the argument. This is useful if you
+    want to tune the values in ``train_loop_config`` as hyperparameters.
+
+    If the ``datasets`` dict contains a training dataset (denoted by
+    the "train" key), then it will be split into multiple dataset
+    shards that can then be accessed by ``ray.train.get_dataset_shard("train")`` inside
+    ``train_loop_per_worker``. All the other datasets will not be split and
+    ``ray.train.get_dataset_shard(...)`` will return the the entire Dataset.
+
+    Inside the ``train_loop_per_worker`` function, you can use any of the
+    :ref:`Ray Train function utils <train-api-func-utils>`.
+
+    .. code-block:: python
+
+        def train_loop_per_worker():
+            # Report intermediate results for callbacks or logging.
+            session.report(...)
+
+            # Checkpoints the provided args as restorable state.
+            session.save_checkpoint(...)
+
+            # Returns dict of last saved checkpoint.
+            session.load_checkpoint()
+
+            # Returns the Ray Dataset shard for the given key.
+            session.get_dataset_shard("my_dataset")
+
+            # Returns the total number of workers executing training.
+            session.get_world_size()
+
+            # Returns the rank of this worker.
+            session.get_world_rank()
+
+            # Returns the rank of the worker on the current node.
+            session.get_local_rank()
+    Args:
+        train_loop_per_worker: The training function to execute.
+            This can either take in no arguments or a ``config`` dict.
+        train_loop_config: Configurations to pass into
+            ``train_loop_per_worker`` if it accepts an argument.
+        jax_config: Configuration for setting up the jax backend. If set to
+            None, use the default configuration. This replaces the ``backend_config``
+            arg of ``DataParallelTrainer``.
+        scaling_config: Configuration for how to scale data parallel training.
+        run_config: Configuration for the execution of the training run.
+        datasets: Any Ray Datasets to use for training. Use
+            the key "train" to denote which dataset is the training
+            dataset. If a ``preprocessor`` is provided and has not already been fit,
+            it will be fit on the training dataset. All datasets will be transformed
+            by the ``preprocessor`` if one is provided.
+        preprocessor: A ``ray.air.preprocessor.Preprocessor`` to preprocess the
```

Comment:
nit: This has been moved to data package.
```suggestion
        preprocessor: A ``ray.data.Preprocessor`` to preprocess the
```